extensions/standard-processors/tests/unit/RollingWindowTests.cpp

```diff
@@ -0,0 +1,108 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string_view>
+#include "Catch.h"
+#include "RollingWindow.h"
+#include "range/v3/view/zip.hpp"
+#include "range/v3/algorithm/contains.hpp"
+
+namespace org::apache::nifi::minifi::test {
+
+using timestamp_type = int;
+using value_type = const char*;
+using RollingWindow = processors::standard::utils::RollingWindow<timestamp_type, value_type>;
+using TimestampComparator = decltype([](const RollingWindow::Entry& lhs, const RollingWindow::Entry& rhs) {
+  return lhs.timestamp < rhs.timestamp;
+});
+
+bool compareEntriesTimestamps(std::vector<RollingWindow::Entry> entries,
+    std::span<const timestamp_type> expected_timestamps) {
+  std::sort(std::begin(entries), std::end(entries), TimestampComparator{});
```

Comment:
`ranges` can do this, too:
```suggestion
  entries |= ranges::actions::sort(TimestampComparator{});
```